### DISCUSSION TITLE
Tabs and Groups shortcut shows on Nightly, not in stable

### DISCUSSION THREAD

**Discussion by JohnLGalt** [menhera-org/TabArray#394] [Q&A] [answered]
Hi,

I've been using CTG for a while now, and I cannot use Fx without. I have it in both profiles, one for Firefox stable and one for Firefox Nightly.

On Nightly, where you can show extension buttons in the browser toolbars, there is an button that links directly to CTG and shows a nice little arrangement of vertically aligned dots with a number showing total number of tabs across all the containers.

![Screenshot 2024-05-02 163953](https://github.com/menhera-org/TabArray/assets/7432512/4f1f5e1c-28c7-4fd7-b16c-9955d03bba18)

In Stable, it is not there. 

Furthermore, I've compared the two profiles side by side, and not only are the CTG settings the same, but virtually all  Firefox Nightly and Firefox Stable settings are the same (I say virtually all, because Nightly does have the Experiments which Stable doesn't have, so those are the only things really different between the two profiles).

I'd like this on Stable as well - any idea what I'm missing? Is it perhaps a setting in About:config that I need to enable?

**Comment by metastable-void** [comment]
I suspect that this is affected by some A/B testing inside Firefox.

Have you recently reinstalled/cleaned up profiles in stable?

Normally, extension buttons need to be added to toolbar/tabbar manually by the user in recent Firefox releases, and the extension manually opting in to place the button in the tab bar has worked around this restriction, but this may have been changed (at least for some users).

**Reply by JohnLGalt** [reply]
Well, I did cleanly install Windows about a month ago, but I copied my profiles first, and after reinstalling both Nightly and Stable, I copied the profiles back (Actually, the exact process is to install both, close both after they create their new default profiles, and delete all existing data files and folders under \AppData\Local\Mozilla and \AppData\Roaming\Mozilla, and copy over the old folders back, so everything is exactly as it was prior to the clean install).

The problem I'm having is that I cannot see a way for me to manually add this back, and while in the Nightly profile I can see a way to remove it, since I don't know how to restore it I've not removed it because I do want it there - I would only remove it if I knew how to restore it, trying to test it. It's not in the customize UI section, it's not in the extensions overflow, it's not in the extensions 'button' (where you can easily manage extensions, the puzzle piece icon).

So, is this something that CTG adds on its own? If so, would you happen to know the last Stable build where this was absolutely working?

**Reply by JohnLGalt** [reply]
The reason I ask is that in neither Firefox Stable nor in Firefox Nightly is there a way to pin CTG - it does not show in the Extensions puzzle piece icon drop down on either browser.

It's just there with the extensions that I have pinned manually from that puzzle piece already in Nightly, and not there in Stable.

I'm going to take a closer look at prej.js and userChrome.css and perhaps a few other files to see if I can spot a difference between the two profiles and try to force it back in Stable.

**Comment by JohnLGalt** [comment, marked as answer]
In comparing extensions.json, I see one difference:

Stable profile:  `"signedState":2,"signedDate":1688711119000,`

Nightly profile: `"signedState":2,"signedTypes":[2,0],"signedDate":1688711119000,`

That signedTypes value is not present in the Stable profile. Does this have anything to do with it being Nightly, and thus having looser restrictions wrt potentially blocked extensions that you can easily allow in Nightly?

I then took a look at pref.js (the one you're not supposed to manually edit).

Stable:

```
user_pref("browser.uiCustomization.state", "{\"placements\":{\"widget-overflow-fixed-list\":[],\"unified-extensions-area\":[\"fdm_ffext2_freedownloadmanager_org-browser-action\",\"_testpilot-containers-browser-action\",\"_contain-facebook-browser-action\",\"enhancerforyoutube_maximerf_addons_mozilla_org-browser-action\",\"_762f9885-5a13-4abd-9c77-433dcd38b8fd_-browser-action\",\"_7b1bf0b6-a1b9-42b0-b75d-252036438bdc_-browser-action\",\"myallychou_gmail_com-browser-action\"],\"nav-bar\":[\"back-button\",\"forward-button\",\"stop-reload-button\",\"urlbar-container\",\"downloads-button\",\"reset-pbm-toolbar-button\",\"_73a6fe31-595d-460b-a920-fcc0f8843232_-browser-action\",\"addon_darkreader_org-browser-action\",\"keepassxc-browser_keepassxc_org-browser-action\",\"forget-me-not_lusito_info-browser-action\",\"unified-extensions-button\",\"fxa-toolbar-menu-button\"],\"toolbar-menubar\":[\"menubar-items\"],\"TabsToolbar\":[\"firefox-view-button\",\"tabbrowser-tabs\",\"new-tab-button\",\"alltabs-button\",\"tab-array_menhera_org-browser-action\"],\"PersonalToolbar\":[\"import-button\",\"personal-bookmarks\"]},\"seen\":[\"save-to-pocket-button\",\"developer-button\",\"_73a6fe31-595d-460b-a920-fcc0f8843232_-browser-action\",\"_contain-facebook-browser-action\",\"enhancerforyoutube_maximerf_addons_mozilla_org-browser-action\",\"_762f9885-5a13-4abd-9c77-433dcd38b8fd_-browser-action\",\"_7b1bf0b6-a1b9-42b0-b75d-252036438bdc_-browser-action\",\"myallychou_gmail_com-browser-action\",\"_testpilot-containers-browser-action\",\"tab-array_menhera_org-browser-action\",\"keepassxc-browser_keepassxc_org-browser-action\",\"fdm_ffext2_freedownloadmanager_org-browser-action\",\"addon_darkreader_org-browser-action\",\"forget-me-not_lusito_info-browser-action\"],\"dirtyAreaCache\":[\"nav-bar\",\"PersonalToolbar\",\"unified-extensions-area\",\"TabsToolbar\",\"toolbar-menubar\",\"widget-overflow-fixed-list\"],\"currentVersion\":20,\"newElementCount\":14}");

```

Nightly:

```
user_pref("browser.uiCustomization.state", "{\"placements\":{\"widget-overflow-fixed-list\":[],\"unified-extensions-area\":[\"fdm_ffext2_freedownloadmanager_org-browser-action\",\"myallychou_gmail_com-browser-action\",\"_7b1bf0b6-a1b9-42b0-b75d-252036438bdc_-browser-action\",\"enhancerforyoutube_maximerf_addons_mozilla_org-browser-action\",\"_762f9885-5a13-4abd-9c77-433dcd38b8fd_-browser-action\",\"regrets-reporter_mozillafoundation_org-browser-action\",\"_contain-facebook-browser-action\",\"_testpilot-containers-browser-action\"],\"nav-bar\":[\"firefox-view-button\",\"back-button\",\"forward-button\",\"stop-reload-button\",\"urlbar-container\",\"downloads-button\",\"tab-array_menhera_org-browser-action\",\"_73a6fe31-595d-460b-a920-fcc0f8843232_-browser-action\",\"addon_darkreader_org-browser-action\",\"keepassxc-browser_keepassxc_org-browser-action\",\"forget-me-not_lusito_info-browser-action\",\"_816dd215-0e91-4621-9d89-3bac78798e6f_-browser-action\",\"unified-extensions-button\",\"reset-pbm-toolbar-button\",\"fxa-toolbar-menu-button\"],\"toolbar-menubar\":[\"menubar-items\"],\"TabsToolbar\":[\"tabbrowser-tabs\",\"new-tab-button\",\"alltabs-button\"],\"PersonalToolbar\":[\"import-button\",\"personal-bookmarks\"]},\"seen\":[\"save-to-pocket-button\",\"profiler-button\",\"developer-button\",\"_73a6fe31-595d-460b-a920-fcc0f8843232_-browser-action\",\"_contain-facebook-browser-action\",\"_testpilot-containers-browser-action\",\"regrets-reporter_mozillafoundation_org-browser-action\",\"_816dd215-0e91-4621-9d89-3bac78798e6f_-browser-action\",\"enhancerforyoutube_maximerf_addons_mozilla_org-browser-action\",\"_762f9885-5a13-4abd-9c77-433dcd38b8fd_-browser-action\",\"_7b1bf0b6-a1b9-42b0-b75d-252036438bdc_-browser-action\",\"myallychou_gmail_com-browser-action\",\"forget-me-not_lusito_info-browser-action\",\"tab-array_menhera_org-browser-action\",\"keepassxc-browser_keepassxc_org-browser-action\",\"addon_darkreader_org-browser-action\",\"fdm_ffext2_freedownloadmanager_org-browser-action\",\"reset-pbm-toolbar-button\"],\"dirtyAreaCache\":[\"nav-bar\",\"PersonalToolbar\",\"toolbar-menubar\",\"TabsToolbar\",\"unified-extensions-area\",\"widget-overflow-fixed-list\"],\"currentVersion\":20,\"newElementCount\":17}");
```

Finally, I decided to bite the bullet and unpin it from Nightly toolbar, and it then shows up in the puzzle piece:

![Screenshot 2024-05-08 133533](https://github.com/menhera-org/TabArray/assets/7432512/5d5fb5a7-7279-4f15-a18a-54c5b9d09923)

However, it is neither pinned nor showing in the puzzle piece in Stable:

![Screenshot 2024-05-08 133732](https://github.com/menhera-org/TabArray/assets/7432512/dd268418-8284-4bb8-b0d8-b33472ed18e4)

And yet the pref.js makes reference to it, seemingly in a _**different**_ (UI) location:

Stable: `\"TabsToolbar\":[\"firefox-view-button\",\"tabbrowser-tabs\",\"new-tab-button\",\"alltabs-button\",\"tab-array_menhera_org-browser-action\"]`

Nightly: `\"nav-bar\":[\"firefox-view-button\",\"back-button\",\"forward-button\",\"stop-reload-button\",\"urlbar-container\",\"downloads-button\",\"tab-array_menhera_org-browser-action\",`

I suspect that this may be a holdover from this profile being recycled countless times from previous Fx versions?

I know that _technically_ it says I should not be manually editing prefs.js, but in this case, do you think it would be worthwhile for me to experiment and do exactly that - move the reference for CTG from the `\TabsToolbar` section to the `\nav-bar` section in the Stable profile?

As a matter of fact, I'm going to try it. Of course, I'll be sure to make backups, and to make sure Firefox is fully closed on my system before editing the file, but I am curious as to why this may be appearing in different sections in the different profiles. Since I'll have the prefs.js files backed up in a completely different folder, I'll be able to restore it if Firefox goes stupid on me.

**Reply by JohnLGalt** [reply]
I'm happy to report that doing this editing of prefs.js fixed it, and it now shows In Firefox Stable using the Stable profile.
